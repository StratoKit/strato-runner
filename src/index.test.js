import test from 'ava'
import path from 'path'

process.chdir('test-project')

test('initial config', t => {
	const config = require('./config').default
	t.snapshot(config)
})

test('initial registry', t => {
	const registry = require('./registry').default
	t.snapshot(registry)
})

test('start', async t => {
	const {start, stop, plugins, config} = require('.').default
	await start('test')
	t.is(plugins.test.started, true)
	t.is(config.state.loadCount, 1)
	t.is(config.state.startCount, 1)
	t.is(config.state2.loadCount, 1)
	t.is(config.state2.startCount, 1)
	t.true(config.promise)
	await stop('test')
	await start('test')
	t.is(config.state2.loadCount, 1)
	t.is(config.state2.startCount, 2)
})
