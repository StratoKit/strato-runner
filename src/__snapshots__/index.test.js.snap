// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial config 1`] = `
Object {
  "appName": "StratoApp",
  "babel": Object {
    "options": Object {
      "plugins": Array [
        "transform-object-rest-spread",
      ],
      "presets": Array [
        Array [
          "env",
          Object {
            "targets": Object {
              "node": "current",
            },
            "useBuiltins": true,
          },
        ],
      ],
    },
    "transpilePlugins": true,
  },
  "languages": Array [
    "en",
  ],
  "npm": Object {
    "dependencies": Object {
      "apollo-client": "^1.2.2",
      "babel-polyfill": "^6.23.0",
      "babel-runtime": "^6.23.0",
      "bcrypt": "^1.0.2",
      "bluebird": "^3.5.0",
      "body-parser": "^1.17.2",
      "compression": "1.6.2",
      "connect-sqlite3": "^0.9.9",
      "cookie-parser": "^1.4.3",
      "cors": "^2.8.3",
      "debug": "^2.6.8",
      "electrode-confippet": "^1.5.0",
      "es6-promisify": "^5",
      "express": "^4.15.3",
      "express-jwt": "^5.3.0",
      "express-session": "^1.15.3",
      "google-libphonenumber": "^2.0.15",
      "graphiql": "^0.10.2",
      "graphql": "^0.9.6",
      "graphql-server-express": "^0.7.2",
      "graphql-tag": "^2.0.0",
      "graphql-type-json": "^0.1.4",
      "i18n-iso-countries": "^1.11.1",
      "in-browser-language": "^1.0.2",
      "isomorphic-fetch": "2.2.1",
      "js-yaml": "*",
      "json5": "^0.5.1",
      "jsurl": "wmertens/jsurl#jsurl2",
      "locale": "^0.1.0",
      "lodash": "^4.17.4",
      "makfy": "^1.1.11",
      "morgan": "^1.8.2",
      "mz": "^2.4.0",
      "react": "^15.5.4",
      "react-addons-perf": "^15.4.2",
      "react-apollo": "^1.2.0",
      "react-deep-force-update": "^2.0.1",
      "react-dom": "^15.5.4",
      "react-keydown": "^1.7.3",
      "react-redux": "^5.0.5",
      "react-router": "^4.1.1",
      "react-router-dom": "^4.1.1",
      "reactabular-column-extensions": "^8.9.3",
      "reactabular-resizable": "^8.9.0",
      "reactabular-table": "^8.9.0",
      "redux": "3.6.0",
      "redux-logger": "^3.0.6",
      "redux-mediaquery": "^0.10.0",
      "redux-persist": "^4.8.0",
      "redux-thunk": "^2.2.0",
      "reselect": "<3.0.0",
      "searchtabular": "^1.5.0",
      "serve-favicon": "^2.4.3",
      "socketcluster": "^5.13.1",
      "socketcluster-client": "^5.5.0",
      "sortabular": "^1.4.0",
      "source-map-support": "^0.4.15",
      "sqlite": "^2.8.0",
      "styled-components": "^2.0.0",
      "table-resolver": "^3.1.0",
      "treetabular": "^3.3.0",
      "uuid": "^3.0.0",
    },
    "devDependencies": Object {
      "assets-webpack-plugin": "^3.5.1",
      "autoprefixer": "^7.1.1",
      "babel-cli": "^6.24.1",
      "babel-core": "^6.24.1",
      "babel-eslint": "^7.2.3",
      "babel-loader": "^7.0.0",
      "babel-plugin-lodash": "^3.2.11",
      "babel-plugin-styled-components": "^1.1.4",
      "babel-plugin-syntax-dynamic-import": "*",
      "babel-plugin-transform-class-properties": "^6.24.1",
      "babel-plugin-transform-decorators-legacy": "^1.3.4",
      "babel-plugin-transform-es2015-destructuring": "^6.23.0",
      "babel-plugin-transform-es2015-parameters": "^6.24.1",
      "babel-plugin-transform-export-extensions": "^6.22.0",
      "babel-plugin-transform-object-rest-spread": "^6.23.0",
      "babel-plugin-transform-react-jsx-self": "^6.22.0",
      "babel-plugin-transform-react-jsx-source": "^6.22.0",
      "babel-plugin-transform-regenerator": "^6.24.1",
      "babel-plugin-transform-runtime": "^6.23.0",
      "babel-plugin-webpack-alias": "^2.1.2",
      "babel-preset-babili": "^0.1.2",
      "babel-preset-env": "^1.5.1",
      "babel-preset-latest": "^6.24.1",
      "babel-preset-react": "^6.24.1",
      "babel-preset-stage-2": "^6.24.1",
      "babel-register": "^6.24.1",
      "concurrently": "^3.4.0",
      "css-loader": "^0.28.4",
      "eslint": "^3.19.0",
      "eslint-config-xo-meatier": "0.2.2",
      "eslint-config-xo-react": "^0.12.0",
      "eslint-import-resolver-webpack": "^0.8.1",
      "eslint-plugin-babel": "^4.1.1",
      "eslint-plugin-import": "^2.3.0",
      "eslint-plugin-react": "^7.0.1",
      "extract-text-webpack-plugin": "^2.1.0",
      "file-loader": "^0.11.1",
      "hard-source-webpack-plugin": "^0.3.12",
      "json-loader": "*",
      "node-object-hash": "^1.2.0",
      "node-sass": "^4.5.3",
      "nodejs-dashboard": "^0.4.1",
      "null-loader": "0.1.1",
      "postcss-loader": "1",
      "postcss-modules-values": "1.2.2",
      "raw-loader": "0.5.1",
      "react-hot-loader": "next",
      "redbox-react": "^1.3.7",
      "rimraf": "^2.6.1",
      "sass-loader": "^6.0.5",
      "start-server-webpack-plugin": "^2.2.0",
      "stats-webpack-plugin": "^0.6.0",
      "style-loader": "^0.18.1",
      "svg-sprite-loader": "^0.3.0",
      "svgo": "^0.7.2",
      "svgo-loader": "^1.2.1",
      "unused-files-webpack-plugin": "^3.0.1",
      "url-loader": "^0.5.8",
      "webpack": "^2.6.1",
      "webpack-bundle-analyzer": "^2.8.2",
      "webpack-cleanup-plugin": "*",
      "webpack-dev-middleware": "^1.10.2",
      "webpack-hot-middleware": "^2.18.0",
      "webpack-node-externals": "^1.6.0",
      "xo": "^0.18.2",
      "yaml-loader": "*",
    },
    "scripts": Object {
      "analyze": "webpack-bundle-analyzer build/client/_webpack_stats.json -m static -r build/client/_webpack_stats.html",
      "analyze-dev": "curl -o build/_webpack_stats-dev.json http://localhost:3000/_/_webpack_stats.json && webpack-bundle-analyzer build/_webpack_stats-dev.json -m static -r build/_webpack_stats-dev.html",
      "bs": "npm run build && npm run prod",
      "build": "concurrently 'npm run build:client' 'npm run build:server'",
      "build-min": "concurrently 'npm run build:client-min' 'npm run build:server-min'",
      "build:client": "NODE_ENV=production webpack --bail --config ./lib/webpack/webpack.config.prod.babel.js",
      "build:client-min": "NODE_ENV=production MINIFY=true webpack --bail --profile --config ./lib/webpack/webpack.config.prod.babel.js",
      "build:dll": "NODE_ENV=development webpack --bail --config ./lib/webpack/webpack.config.dll.babel.js",
      "build:server": "NODE_ENV=production webpack --bail --config ./lib/webpack/webpack.config.server.prod.babel.js",
      "build:server-min": "NODE_ENV=production MINIFY=true webpack --bail --profile --config ./lib/webpack/webpack.config.server.prod.babel.js",
      "clear-cache": "rimraf build/cache ~/.babel.json",
      "dash:dev": "NODEJS_DASH=true nodejs-dashboard -- npm run start:dev",
      "dash:prod": "NODEJS_DASH=true nodejs-dashboard -- npm run start:prod",
      "lint": "xo 'src/**/*.js' 'lib/**/*.js' 'src/**/*.jsx' 'lib/**/*.jsx'",
      "lint-fix": "xo --fix 'src/**/*.js' 'lib/**/*.js' 'src/**/*.jsx' 'lib/**/*.jsx'",
      "makfy": "node -r babel-register ./node_modules/.bin/makfy",
      "prod": "npm run dash:prod",
      "start": "node -r babel-register ./node_modules/.bin/makfy startDev",
      "start-inspect": "node -r babel-register ./node_modules/.bin/makfy startDev --inspect",
      "start:dev": "NODE_ENV=development webpack --config ./lib/webpack/webpack.config.server.babel.js --watch",
      "start:dev-client": "NODE_ENV=development webpack --config ./lib/webpack/webpack.config.dev.babel.js --watch",
      "start:prod": "NODE_ENV=production node ./build/server/server.js",
      "writeConfig": "node -r babel-register ./node_modules/.bin/makfy writeConfig",
    },
    "version": "{{config.version}}",
  },
  "paths": Object {
    "app": "{{config.paths.root}}/src",
    "assets": "{{config.paths.root}}/assets",
    "build": Object {
      "cache": "{{config.paths.root}}/build/cache",
      "client": "{{config.paths.root}}/build/client",
      "dll": "{{config.paths.root}}/build/dll",
      "root": "{{config.paths.root}}/build",
      "server": "{{config.paths.root}}/build/server",
    },
    "icons": "{{config.paths.assets}}/icons",
    "nodeModules": "{{config.paths.root}}/node_modules",
    "root": "{{cwd}}",
    "state": "{{config.paths.app}}/redux",
    "stratokit": "{{config.paths.root}}/lib",
    "upload": "{{config.paths.build.root}}/upload",
  },
  "sections": Object {
    "a": "APP",
    "c": "COUNTRY",
  },
  "stratokit": Object {
    "graphiql": true,
    "plugins": Array [],
    "server": Object {
      "brokers": 1,
      "entry": "stratokit/server",
      "host": "localhost",
      "logLevel": 1,
      "port": 3000,
      "requireBasicAuth": null,
      "socketChannelLimit": 1000,
      "workers": 1,
    },
    "sessions": Object {
      "dir": "{{config.paths.build.root}}",
    },
    "stage": "CONFIG",
  },
  "testing": 123,
  "version": "0.1.0",
  "webpack": Object {
    "alias": Object {
      "app": "{{config.paths.app}}",
      "assets": "{{config.paths.assets}}",
      "lodash-es": "lodash",
      "lodash._getnative": "lodash/_getnative",
      "lodash.flatten": "lodash/flatten",
      "lodash.isarguments": "lodash/isArguments",
      "lodash.isarray": "lodash/isArray",
      "lodash.keys": "lodash/keys",
      "lodash.pick": "lodash/pick",
      "state": "{{config.paths.state}}",
      "stratokit": "{{config.paths.stratokit}}",
    },
    "defines": Object {
      "__CONFIG__": Object {
        "graphiql": "{{config.stratokit.graphiql}}",
        "languages": "{{config.languages}}",
        "name": "{{config.appName}}",
        "sections": "{{config.sections}}",
        "version": "{{config.version}}",
      },
      "__IN_WEBPACK__": true,
    },
    "extensions": Object {
      "assets": Array [
        "png",
        "jpg",
        "jpeg",
        "gif",
        "ttf",
        "eot",
        "woff",
        "woff2",
        "mp3",
        "wav",
        "svg",
      ],
      "css": Array [
        "css",
        "scss",
        "sass",
      ],
      "js": Array [
        "js",
        "jsx",
      ],
      "json": Array [
        "json",
      ],
      "raw": Array [
        "txt",
      ],
      "sass": Array [
        "scss",
        "sass",
      ],
      "svg": Array [
        "svg",
      ],
      "yaml": Array [
        "yaml",
      ],
    },
    "loaderConfig": Object {},
    "loaders": Array [],
    "publicPath": "/_",
    "use": Object {
      "css": Object {
        "autoprefixer": true,
        "enabled": true,
        "modules": true,
        "sass": true,
      },
      "svgo": Object {
        "enabled": true,
      },
      "urlLoader": 1000,
    },
  },
}
`;

exports[`initial registry 1`] = `
Object {
  "name": Object {
    "config": Object {
      "name": true,
    },
    "name": "name",
  },
  "promise": Promise {},
  "test": Object {
    "config": Object {
      "state": Object {
        "loadCount": 0,
        "startCount": 0,
      },
    },
    "load": [Function],
    "name": "test",
    "requires": Array [
      "test2",
    ],
    "start": [Function],
    "version": "1.0.0",
  },
  "test2": Object {
    "config": Object {
      "state2": Object {
        "loadCount": 0,
        "startCount": 0,
      },
    },
    "load": [Function],
    "name": "test2",
    "requires": Array [
      "promise",
    ],
    "start": [Function],
  },
}
`;
